File: PatronesGOF/src/ds/patrones/singleton/ImpresoraSingleton.java
package ds.patrones.singleton;

public class ImpresoraSingleton {
	
	private static ImpresoraSingleton instancia;
	
	public static ImpresoraSingleton getInstancia() {
		
		if (instancia == null) {
			instancia = new ImpresoraSingleton();
		}
		return instancia;
	}
	
	public void imprimirTrabajo(String mensaje) {
		System.out.println(mensaje);
	}

}

File: PatronesGOF/src/ds/patrones/singleton/python/ImpresoraSingleton.py
class ImpresoraSingleton():

    __instancia = None

    def __init__(selft):
        pass

    def __new__(cls):
        if ImpresoraSingleton.__instancia is None:
            ImpresoraSingleton.__instancia= object.__new__(cls)

        return ImpresoraSingleton.__instancia

    def imprimirTrabajo(self,mensaje):
        print(mensaje)